### PR TITLE
Add .titleCase() revisited

### PR DESCRIPTION
**Pull request by pbihler**
Performed the modifications requested in #88 and added documentation.
Replaces PR #88

### CODE REVIEW

**Comment by jprichardson** [comment]
Looks good to me. @az7arul when you review it, feel free to merge and publish.

File: README.md [hunk 1]
```diff
@@ -515,7 +515,7 @@ Example:
 var stuff = "My name is JP\nJavaScript is my fav language\r\nWhat is your fav language?"
 var lines = S(stuff).lines()
 
-console.dir(lines) 
+console.dir(lines)
```
Comment (az7arul) [inline README.md]: unnecessary diffs

File: README.md [hunk 2]
```diff
@@ -526,7 +526,7 @@ console.dir(lines)
 
 ### - pad(len, [char])
 
-Pads the string in the center with specified character. `char` may be a string or a number, defaults is a space. 
+Pads the string in the center with specified character. `char` may be a string or a number, defaults is a space.
```
Comment (az7arul) [inline README.md]: unnecessary diffs

File: README.md [hunk 3]
```diff
@@ -979,7 +997,7 @@ S('&lt;div&gt;hi&lt;/div&gt;').unescapeHTML().s; //<div>hi</div>
 
 ### - wrapHTML() ###
 
-wrapHTML helps to avoid concatenation of element with string. 
+wrapHTML helps to avoid concatenation of element with string.
```
Comment (az7arul) [inline README.md]: unnecessary diff

File: README.md [hunk 4]
```diff
@@ -1083,7 +1101,7 @@ If you contribute to this library, just modify `string.js`, `string.test.js`, an
 - [*] [Nathan Friedly](https://github.com/nfriedly)
 - [*] [Alison Rowland](https://github.com/arowla)
 
- 
+
```
Comment (az7arul) [inline README.md]: unnecessary diff

File: README.md [hunk 5]
```diff
@@ -1116,6 +1134,3 @@ THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLI
 [aboutjp]: http://about.me/jprichardson
```
Comment (az7arul) [inline README.md]: unnecessary diff

File: test/string.test.js [hunk 6]
```diff
@@ -545,7 +545,7 @@
 
         var str = "{{greet }} {{ name}}! How are you doing during the year of {{  date-year }}?";
         EQ (S(str).template(values).s, 'Hello JP! How are you doing during the year of 2013?')
-						
+
```
Comment (az7arul) [inline test/string.test.js]: unnecessary diff

**Comment by az7arul** [comment]
@pbihler thanks, can you  squash these commits,  remove the unnecessary diffs and rebase ?
